File: Q2_InsertInMiddle.c
#include <stdio.h>
#include <stdlib.h>
struct Node {int data; struct Node* next;};
void insert_mid(struct Node* prev,int val){if(!prev)return;struct Node* n=malloc(sizeof(struct Node));n->data=val;n->next=prev->next;prev->next=n;}
void print(struct Node* h){while(h){printf("%d ",h->data);h=h->next;}}
int main(){struct Node* head=malloc(sizeof(struct Node));head->data=1;head->next=malloc(sizeof(struct Node));head->next->data=3;head->next->next=NULL;insert_mid(head,2);print(head);}

File: Q2_InsertInMiddle.py
class Node:
    def __init__(self,data):
        self.data=data
        self.next=None
def insert_mid(prev,val):
    if not prev:return
    n=Node(val)
    n.next=prev.next
    prev.next=n
def print_list(h):
    while h:
        print(h.data,end=" ")
        h=h.next
head=Node(1)
head.next=Node(3)
insert_mid(head,2)
print_list(head)